MyRangesImplementation/MyRangesImplementation/filter_range_adapter.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+#include"range_traits.hpp"
+#include"filter_iterator.hpp"
+#include<type_traits>
+#include<vector>
+
+namespace ranges {
+	namespace view {
+		template<typename Range, typename FilterFunc>
+		class filter_range_adapter {
```

Comment:
I feel like all your adapters follow the same concept.

is there any way to enforce and explain that in code?